python/pyspark/sql/pandas/types.py

```diff
@@ -171,7 +171,7 @@ def to_arrow_type(
     elif type(dt) == VariantType:
         fields = [
             pa.field("value", pa.binary(), nullable=False),
-            pa.field("metadata", pa.binary(), nullable=False),
+            pa.field("metadata", pa.binary(), nullable=False, metadata={b"variant": b"true"}),
```

Comment:
Should we comment on the scheme we are using to "tag" an arrow struct as a variant?

It looks like we are attaching some metadata to the `metadata` field of the struct?